File: wholeNum.java
public class WholeNum {
    private static final long UINT_MAX = (1L << 32) - 1;
    private final long value;

    public WholeNum(long value) {
        this.value = Math.min(Math.max(value, 0), UINT_MAX);
    }

    public long getValue() {
        return value;
    }

    public WholeNum getSuccessor() {
        return new WholeNum(value == UINT_MAX ? UINT_MAX : value + 1);
    }

    public WholeNum add(WholeNum other) {
        return new WholeNum(Math.min(value + other.value, UINT_MAX));
    }

    public WholeNum subtract(WholeNum other) {
        return new WholeNum(Math.max(value - other.value, 0));
    }

    public boolean equals(WholeNum other) {
        return this.value == other.value;
    }

    @Override
    public String toString() {
        return String.valueOf(value);
    }

    public static void main(String[] args) {
        WholeNum zero = new WholeNum(0);
        WholeNum one = zero.getSuccessor();
        WholeNum sum = one.add(one);

        System.out.println(zero); // 0
        System.out.println(one); // 1
        System.out.println(sum); // 2

        if (zero.equals(new WholeNum(0))) {
            System.out.println("0 is zero");
        }

        if (one.equals(sum)) {
            System.out.println("Something is wrong: 1 is equal to sum");
        } else {
            System.out.println("Correct: 1 is not equal to sum (1 != 2)");
        }
    }
}

File: wholeNum.py
class WholeNum:
    UINT_MAX = (1 << 32) - 1  # Simulating unsigned int max value

    def __init__(self, value=0):
        self.value = min(max(value, 0), WholeNum.UINT_MAX)

    def get_value(self):
        return self.value

    def get_successor(self):
        return WholeNum(self.value + 1 if self.value < WholeNum.UINT_MAX else WholeNum.UINT_MAX)

    def __add__(self, other):
        return WholeNum(min(self.value + other.value, WholeNum.UINT_MAX))

    def __sub__(self, other):
        return WholeNum(max(self.value - other.value, 0))

    def __eq__(self, other):
        return self.value == other.value

    def __str__(self):
        return str(self.value)


if __name__ == "__main__":
    zero = WholeNum()
    one = zero.get_successor()
    sum_val = one + one

    print(zero)  # 0
    print(one)  # 1
    print(sum_val)  # 2

    if zero == WholeNum(0):
        print("0 is zero")

    if one == sum_val:
        print("Something is wrong: 1 is equal to sum")
    else:
        print("Correct: 1 is not equal to sum (1 != 2)")
